src/string_impls/to_string.rs

```diff
@@ -13,16 +13,358 @@
+    ///
+    /// to_writer_custom()
+    ///     .compact()
+    ///     .write(&val, &mut s)
+    ///     .unwrap();
+    ///
+    /// assert_eq!(s, r#"{foo:(1,2,3,4,5),bar:true}"#);
+    /// ```
+    pub fn compact(mut self) -> Self {
+        self.style = FormatStyle::Compact;
+        self
+    }
+
+    /// Write the [`crate::Value`] to a pretty spaced string.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use scale_value::{value, Value};
+    /// use scale_value::stringify::to_writer_custom;
+    ///
+    /// let val = value!({
+    ///     foo: (1,2,3,4,5),
+    ///     bar: true
+    /// });
+    ///
+    /// let mut s = String::new();
+    ///
+    /// to_writer_custom()
+    ///     .spaced()
+    ///     .write(&val, &mut s)
+    ///     .unwrap();
+    ///
+    /// assert_eq!(s, r#"{
+    ///   foo: (
+    ///     1,
+    ///     2,
+    ///     3,
+    ///     4,
+    ///     5
+    ///   ),
+    ///   bar: true
+    /// }"#);
+    /// ```
+    pub fn spaced(mut self) -> Self {
+        self.style = FormatStyle::Indented(0);
+        self
+    }
+
+    /// Write the [`crate::Value`] to a pretty spaced string, indenting
+    /// each new line by applying [`Self::indent_by()`] the number of
+    /// times given here. Defaults to nothing.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use scale_value::{value, Value};
+    /// use scale_value::stringify::to_writer_custom;
+    ///
+    /// let val = value!({
+    ///     foo: (1,2,3,4,5),
+    ///     bar: true
+    /// });
+    ///
+    /// let mut s = String::new();
+    ///
+    /// to_writer_custom()
+    ///     .spaced()
+    ///     .leading_indent("****")
+    ///     .write(&val, &mut s)
+    ///     .unwrap();
+    ///
+    /// assert_eq!(s, r#"{
+    /// ****  foo: (
+    /// ****    1,
+    /// ****    2,
+    /// ****    3,
+    /// ****    4,
+    /// ****    5
+    /// ****  ),
+    /// ****  bar: true
+    /// ****}"#);
+    /// ```
+    pub fn leading_indent(mut self, s: impl Into<String>) -> Self {
+        self.leading_indent = s.into();
+        self
+    }
+
+    /// When using [`Self::spaced()`], this defines the characters used
+    /// to add each step of indentation to newlines. Defaults to
+    /// two spaces.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use scale_value::{value, Value};
+    /// use scale_value::stringify::to_writer_custom;
+    ///
+    /// let val = value!({
+    ///     foo: (1,2,3,4,5),
+    ///     bar: true
+    /// });
+    ///
+    /// let mut s = String::new();
+    ///
+    /// to_writer_custom()
+    ///     .spaced()
+    ///     .indent_by("**")
+    ///     .write(&val, &mut s)
+    ///     .unwrap();
+    ///
+    /// assert_eq!(s, r#"{
+    /// **foo: (
+    /// ****1,
+    /// ****2,
+    /// ****3,
+    /// ****4,
+    /// ****5
+    /// **),
+    /// **bar: true
+    /// }"#);
+    /// ```
+    pub fn indent_by(mut self, s: impl Into<String>) -> Self {
```

Comment:
what happens for non-symmetrical prefix patterns? 
ie: `example >: <body>`

Reply:
you'd end up applying it for each level of indentation so you'd get `example >: exampkle >:` or whatever for 2 levels of indent, if that's what you mean?

Reply:
Oh, ok. Missed that bit